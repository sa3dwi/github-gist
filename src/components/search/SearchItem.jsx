import React from 'react';

const SearchItem = (props) => {

    return (
        <li className="list-item">
            <div>
                <span className="strong7">Description: </span> <p> {props.item.description} </p><br/>
                <span className="strong7">Comments: </span>{props.item.comments} <br/>
                <span className="strong7">Created at: </span>{props.item.created_at} <br/>
                <span className="strong7">Updated at: </span>{props.item.updated_at} <br/>
                <ul className="flex-list flex-item">
                    {typeof(props.item.files) !== 'undefined' ?
                        props.item.files.length > 0 && props.item.files.map((file, index) => {
                        return (
                            <div className="flex-item" key={index}>
                                <img width="32" height="32" alt={file.language} src={`./../../../images/badges/${file.language.toLowerCase()}.png`} 
                                onError={(e)=>{e.target.onerror = null; e.target.src="./../../../images/badges/default.png"}} />
                            </div>
                        );
                    }): ''}
                </ul> 
                <ul className="flex-list flex-item">
                    <img width="16" height="16" alt="forkList" src="./../../../images/fork.png"/>
                    {typeof(props.item.forks) === 'undefined' || props.item.forks.length === 0 ? 
                        <span>
                            No forks
                        </span> 
                        :
                        props.item.forks.length > 0 && props.item.forks.map((fork, index) => {
                            return (
                                <a key={index} target="_blank" className="flex-item" rel="noopener noreferrer" href={fork.html_url}>
                                    <img className="user_avatar" width="64" height="64" alt={fork.username} src={fork.avatar_url}
                                        onError={(e)=>{e.target.onerror = null; e.target.src="./../../../images/gitHub_logo.png"}} /> {fork.username}</a>                               
                            );
                        })
                    }
                </ul> 
            </div>
            <div>
                <a target="_blank" rel="noopener noreferrer" href={props.item.html_url}>Browse Gist page</a>
            </div>
        </li>
    );
    
};

export default SearchItem;