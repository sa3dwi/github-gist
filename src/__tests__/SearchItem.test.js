import React from 'react';
import {render, cleanup} from '@testing-library/react'
import { create } from "react-test-renderer";
import { expect } from 'chai';

import SearchItem from './../components/search/SearchItem';

beforeEach(() => {
  localStorage.clear();
  cleanup
})

describe("SearchItem component UI", () => {
  test('matches the snapshot', () => {
    const wrapper = create(<SearchItem item={{item:[]}}/>)
    expect(wrapper.toJSON()).to.matchSnapshot()
  })
})

// TODO: use mock api 
describe("SearchItem component Functionality", () => {
  test('allows you show search item', () => {
    render(<SearchItem item={{item:[]}}/>)
  })
})
