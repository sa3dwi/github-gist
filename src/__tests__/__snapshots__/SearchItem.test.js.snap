// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchItem component UI matches the snapshot 1`] = `
<li
  className="list-item"
>
  <div>
    <span
      className="strong7"
    >
      Description: 
    </span>
     
    <p>
       
       
    </p>
    <br />
    <span
      className="strong7"
    >
      Comments: 
    </span>
     
    <br />
    <span
      className="strong7"
    >
      Created at: 
    </span>
     
    <br />
    <span
      className="strong7"
    >
      Updated at: 
    </span>
     
    <br />
    <ul
      className="flex-list flex-item"
    >
      
    </ul>
    <ul
      className="flex-list flex-item"
    >
      <img
        alt="forkList"
        height="16"
        src="./../../../images/fork.png"
        width="16"
      />
      <span>
        No forks
      </span>
    </ul>
  </div>
  <div>
    <a
      rel="noopener noreferrer"
      target="_blank"
    >
      Browse Gist page
    </a>
  </div>
</li>
`;
